openedx/core/djangoapps/profile_images/tests/test_views.py

```diff
@@ -405,13 +385,79 @@ def test_remove_staff(self, mock_log):
     @patch('student.models.UserProfile.save')
     def test_remove_failure(self, user_profile_save, mock_log):
         """
-        Test that when upload validation fails, the proper HTTP response and
+        Test that when remove validation fails, the proper HTTP response and
         messages are returned.
         """
         user_profile_save.side_effect = [Exception(u"whoops"), None]
         with self.assertRaises(Exception):
-            self.client.post(self.url)
+            self.client.delete(self.url)
         self.check_images(True)  # thumbnails should remain intact.
         self.check_has_profile_image(True)
         self.assertFalse(mock_log.info.called)
         self.assert_no_events_were_emitted()
+
+
+@unittest.skipUnless(settings.ROOT_URLCONF == 'lms.urls', 'Profile Image API is only supported in LMS')
+@mock.patch('openedx.core.djangoapps.profile_images.views.log')
+class DeprecatedProfileImageUploadTestCase(ProfileImageEndpointTestCase):
+    """
+    Tests for the deprecated profile_image upload endpoint.
+    """
+
+    _view_name = 'profile_image_upload'
+
+    # Use the patched version of the API client to workaround a unicode issue
+    # with DRF 3.1 and Django 1.4.  Remove this after we upgrade Django past 1.4!
+    client_class = PatchedClient
+
+    def test_unsupported_methods(self, mock_log):
+        """
+        Test that GET, PUT, PATCH, and DELETE are not supported.
+        """
+        self.assertEqual(405, self.client.get(self.url).status_code)
+        self.assertEqual(405, self.client.put(self.url).status_code)
+        self.assertEqual(405, self.client.patch(self.url).status_code)
+        self.assertEqual(405, self.client.delete(self.url).status_code)
+        self.assertFalse(mock_log.info.called)
+        self.assert_no_events_were_emitted()
+
+    @patch('openedx.core.djangoapps.profile_images.views.ProfileImageView.post')
+    def test_post_calls_replacement_view_post(self, mock_post, mock_log):
+        mock_post.return_value = HttpResponse()
+        self.client.post(self.url)
+        assert mock_post.called
+
+
+@unittest.skipUnless(settings.ROOT_URLCONF == 'lms.urls', 'Profile Image API is only supported in LMS')
+@mock.patch('openedx.core.djangoapps.profile_images.views.log')
+class DeprecatedProfileImageRemoveTestCase(ProfileImageEndpointTestCase):
```

Comment:
Consider creating a common test Mixin class for `DeprecatedProfileImageUploadTestCase` and `DeprecatedProfileImageRemoveTestCase` to reduce code duplication.